buildpacks/nodejs-function-invoker/src/layers/runtime.rs

```diff
@@ -0,0 +1,106 @@
+use crate::{NodeJsInvokerBuildpack, NodeJsInvokerBuildpackError};
+use libcnb::build::BuildContext;
+use libcnb::data::buildpack::StackId;
+use libcnb::data::layer_content_metadata::LayerTypes;
+use libcnb::layer::{ExistingLayerStrategy, Layer, LayerData, LayerResult, LayerResultBuilder};
+use libcnb::Buildpack;
+use libherokubuildpack::log_info;
+use serde::{Deserialize, Serialize};
+use std::io::{self, Write};
+use std::path::Path;
+use std::process::Command;
+use thiserror::Error;
+
+/// A layer that installs the Node.js Invoker/Runtime package
+pub struct RuntimeLayer {
+    pub package: String,
+}
+
+#[derive(Deserialize, Serialize, Clone, PartialEq, Eq)]
+pub struct RuntimeLayerMetadata {
+    layer_version: String,
+    package: String,
+    stack_id: StackId,
+}
+
+#[derive(Error, Debug)]
+pub enum RuntimeLayerError {
+    #[error("Couldn't run `npm install` command: {0}")]
+    NpmCommandError(std::io::Error),
+    #[error("Couldn't install invoker runtime with `npm install`: #{0}")]
+    NpmInstallError(std::process::ExitStatus),
+}
+
+const LAYER_VERSION: &str = "1";
+
+impl Layer for RuntimeLayer {
+    type Buildpack = NodeJsInvokerBuildpack;
+    type Metadata = RuntimeLayerMetadata;
+
+    fn types(&self) -> LayerTypes {
+        LayerTypes {
+            build: true,
+            launch: true,
+            cache: true,
+        }
+    }
+
+    fn create(
+        &self,
+        context: &BuildContext<Self::Buildpack>,
+        layer_path: &Path,
+    ) -> Result<LayerResult<Self::Metadata>, NodeJsInvokerBuildpackError> {
+        log_info(format!(
+            "Installing Node.js Function Invoker Runtime {}",
+            self.package
+        ));
+
+        Command::new("npm")
+            .args([
+                "install",
+                "-g",
+                "--prefix",
+                &layer_path.to_string_lossy(),
+                &self.package,
+            ])
+            .output()
+            .map_err(RuntimeLayerError::NpmCommandError)
+            .and_then(|output| {
+                if output.status.success() {
+                    Ok(output)
+                } else {
+                    io::stdout().write_all(&output.stdout).ok();
+                    io::stderr().write_all(&output.stderr).ok();
+                    Err(RuntimeLayerError::NpmInstallError(output.status))
+                }
+            })?;
```

Comment:
Nit: While this works, explicitly writing stdout/err can be omitted if we use `spawn` instead of `output` which will pipe stdout and stderr to the parent process (this buildpack).

```suggestion
            .spawn()
            .and_then(|mut child| child.wait())
            .map_err(RuntimeLayerError::NpmCommandError)
            .and_then(|status| {
                status.success()
                    .then(|| ())
                    .ok_or( RuntimeLayerError::NpmInstallError(status))
            })?;
```

Reply:
I think this would always send stdout and stderr back to the build log? These logs are noisy and uninteresting in the happy path, so I'm trying to suppress them for successful installs.

Reply:
Ah, I totally missed that detail, oops. Nevermind then, sorry :) Might be worth a comment in the source?

Reply:
Added a comment and also dropped the `if {} else {}` in favor of a `.then()`, which I think may have been an implicit gripe. =P